.github/workflows/tests.yaml

```diff
@@ -132,6 +132,40 @@ jobs:
               uses: codecov/codecov-action@v1
               if: matrix.python-version == '3.7'
 
+    cockroach:
+        runs-on: ubuntu-latest
+        timeout-minutes: 30
+        strategy:
+            matrix:
+                python-version: ["3.7", "3.8", "3.9", "3.10"]
+                cockroachdb-version: ["v22.2.0-beta.1"]
+        steps:
+            - uses: actions/checkout@v3
+            - name: Set up Python ${{ matrix.python-version }}
+              uses: actions/setup-python@v3
+              with:
+                  python-version: ${{ matrix.python-version }}
+            - name: Install dependencies
+              run: |
+                  python -m pip install --upgrade pip
+                  pip install -r requirements/requirements.txt
+                  pip install -r requirements/test-requirements.txt
+                  pip install -r requirements/extras/postgres.txt
+            - name: Setup CockroachDB
+              run: |
+                  wget -qO- https://binaries.cockroachdb.com/cockroach-${{ matrix.cockroachdb-version }}.linux-amd64.tgz | tar xz
+                  ./cockroach-${{ matrix.cockroachdb-version }}.linux-amd64/cockroach start-single-node --insecure --background
+                  ./cockroach-${{ matrix.cockroachdb-version }}.linux-amd64/cockroach sql --insecure -e 'create database piccolo;'
```

Comment:
When running Cockroach locally with `cockroach start-single-node --insecure` I get quite slow performance. Should we pass in any extra flags here to give it more resources?

Reply:
`./cockroach start-single-node --insecure --background` should give Cockroach uncapped resources.

The poor performance is likely just due to warmup / node generation. Although you could try adding `--store=node_1` to ensure it's writing to the local disk beside the binary. The tests on Github are already pretty fast though, I wouldn't worry about it (only a few minutes to run).

You can point Cockroach `--store=` to anything writable and it will use it as the database.

For best performance, I just keep a binary of Cockroach around that I can start at any time (it will re-use the on-disk state when available, in my case `./node_1`... which just looks like a folder beside the binary where Cockroach is ran).

Full reference here: https://www.cockroachlabs.com/docs/dev/cockroach-start-single-node.html

Reply:
Cool, thanks for the explanation.